posthog/tasks/process_event.py

```diff
@@ -136,15 +139,7 @@ def _capture(
             for index, el in enumerate(elements)
         ]
 
-    team = Team.objects.only(
-        "slack_incoming_webhook",
-        "event_names",
-        "event_properties",
-        "event_names_with_usage",
-        "event_properties_with_usage",
-        "anonymize_ips",
-        "ingested_event",
-    ).get(pk=team_id)
+    team = Team.objects.get_team_cached(pk=team_id)
```

Comment:
Wouldn't this cause issues if user updates any of the following properties? These would be cached and user would still receive PII/no slack hooks/wrong event info until they restart the server.

```
        "slack_incoming_webhook",	
        "event_names",	
        "event_properties",	
        "event_names_with_usage",	
        "event_properties_with_usage",	
        "anonymize_ips",	
        "ingested_event",
```

Reply:
Yep, that's fair. Some of those would have been correctly re-checked, but i've gone for a redis backed approach now instead.

Question for @fuziontech. Will using redis even help or is this just shifting the problem?

Reply:
We have the post save trigger on the team object that would update the cache immediately right? Seems like it would work nicely.

Hard problems of computing though - cache invalidation.